### ISSUE TITLE
Can not download S3 object in Lambda's execution time with latest tag localstack image

### ISSUE THREAD

**Issue by dinhnguyen7025** [localstack/localstack#3160]
# Type of request:
* [x] bug report

# Detailed description
* When I try push message to SQS queue and invoke lambda function, everything is fine, and in lambda function I need to download file from s3 which is also defined on localstack too. But I get an error.

## Expected behavior
* Download file from s3 successfully in lambda function

## Actual behavior
```shell
localstack_1  | 2020-10-23T03:22:15:WARNING:localstack.utils.server.http2_server: Error in proxy handler for request GET http://localhost/bucket-name/test_file.txt: An error occurred (404) when calling the HeadObject operation: Not Found Traceback (most recent call last):
localstack_1  |   File "/opt/code/localstack/localstack/utils/server/http2_server.py", line 94, in index
...
localstack_1  | botocore.exceptions.ClientError: An error occurred (404) when calling the HeadObject operation: Not Found
localstack_1  |   2020-10-23T03:22:17:DEBUG:localstack.services.awslambda.lambda_executors: Lambda arn:aws:lambda:us-east-1:000000000000:function: LambdaFunction result / log output:
localstack_1  | null
localstack_1  | > START RequestId: 0f7186c8-a80e-1a12-41df-3a8de4ff75f2 Version: $LATEST
localstack_1  | > 2020/10/23 03:22:15 InternalServerError: Internal Server Error
localstack_1  | > 	status code: 500, request id: , host id: 
localstack_1  | > 2020/10/23 03:22:15 Downloaded /tmp/artifact/1603423335272966500/test_file.txt 0 bytes
localstack_1  | > 2020/10/23 03:22:15 /tmp/artifact/1603423335272966500/test_file.txt
localstack_1  | > END RequestId: 0f7186c8-a80e-1a12-41df-3a8de4ff75f2
localstack_1  | > REPORT RequestId: 0f7186c8-a80e-1a12-41df-3a8de4ff75f2	Init Duration: 2097.35 ms	Duration: 559.82 ms	Billed Duration: 600 ms	Memory Size: 1536 MB	Max Memory Used: 28 MB
```

# Steps to reproduce
* Create simple stack with SQS queue and lambda function as template.yaml

```
AWSTemplateFormatVersion: '2010-09-09'
Transform: AWS::Serverless-2016-10-31

Resources:
  LambdaFunction:
    Type: AWS::Serverless::Function
    Properties:
      FunctionName: "LambdaFunction"
      CodeUri:
        Bucket: bucket-name
        Key: file-name
      Handler: bin/main
      Runtime: go1.x
      Tracing: Active # https://docs.aws.amazon.com/lambda/latest/dg/lambda-x-ray.html
      Policies:
        - Version: '2012-10-17' # Policy Document
          Statement:
            - Effect: Allow
              Action:
                - s3:*
              Resource: 'arn:aws:s3:::bucket-name/*'
      Events:
        MySQSEvent:
          Type: SQS
          Properties:
            Queue: !GetAtt SqsQueue.Arn

  SqsQueue:
    Type: AWS::SQS::Queue
    Properties:
      QueueName: "SqsQueue"
```

* Apply stack with awslocal command:

```shell
awslocal cloudformation create-stack \
  --stack-name simple-stack \
  --template-body file://template.yaml \
  --capabilities CAPABILITY_IAM
```

## Client code (AWS SDK code snippet)
* Golang aws sdk

```golang
import (
	"bytes"
	"context"
	"fmt"
	"github.com/aws/aws-sdk-go/aws/credentials"
	"github.com/aws/aws-sdk-go/service/s3/s3manager"
	"io"
	"io/ioutil"
	"log"
	"os"
	"regexp"
	"strconv"
	"time"

	"github.com/aws/aws-lambda-go/events"
	"github.com/aws/aws-sdk-go/service/s3"

	"github.com/aws/aws-sdk-go/aws"
	"github.com/aws/aws-sdk-go/aws/session"
        "github.com/aws/aws-sdk-go/service/sqs"
)




func main(ctx context.Context, sqsEvent events.SQSEvent) error {
       sess := session.Must(session.NewSession(&aws.Config{
		S3ForcePathStyle: aws.Bool(true),
		Endpoint:         aws.String("http://host.docker.internal:4566"),
	}))

       s3Downloader := s3manager.NewDownloader(sess)

	//create empty folder
	now := strconv.Itoa(int(time.Now().UnixNano()))
	tmpDir := "/tmp/artifact/" + now + "/"
	if _, err := os.Stat(tmpDir); err == nil {
		if err := os.RemoveAll(tmpDir); err != nil {
			return err
		}
	}

	if err := os.MkdirAll(tmpDir, 0755); err != nil {
		return err
	}


	file, err := os.Create(tmpDir + "test_file.txt")
	if err != nil {
		return err
	}

	defer file.Close()

	numBytes, err := s3Downloader.Download(file, &s3.GetObjectInput{
		Bucket: aws.String("bucket-name"),
		Key:    aws.String("test_file.txt"),
	})
	if err != nil {
		log.Println(err)
	}

	log.Println("Downloaded", file.Name(), numBytes, "bytes")
}
```

┆Issue is synchronized with this [Jira Task](https://localstack.atlassian.net/browse/LOC-83) by [Unito](https://www.unito.io/learn-more)

**Comment by usmangani1** [comment]
@dinhnguyen7025  Thanks for reporting the issue.
Can you please help us with the working code using which we can reproduce this issue.
There are a lot of import errors when I tired to build the code which was given above.
Thanks

**Comment by dinhnguyen7025** [comment]
@usmangani1 Hi, Thank for response. I make a [repo](https://github.com/dinhnguyen7025/example-using-sqs-with-localstack) to reproduce this issue. Please check and tell me if there is any import errors exist. Btw, the following is log from localstack when I try to get file from s3.

> 2020-11-30T03:41:39:WARNING:localstack.utils.server.http2_server: Error in proxy handler for request GET http://localhost/custom-bucket/Book1.xlsx: An error occurred (404) when calling the HeadObject operation: Not Found Traceback (most recent call last):

**Comment by whummer** [comment]
Thanks for reporting @dinhnguyen7025 . This should be fixed in #3370 - can you please pull the `latest` Docker image and give it another try? Please report here if the problem persists. Thanks
